ir/instr.cpp

```diff
@@ -227,25 +232,57 @@ StateValue BinOp::toSMT(State &s) const {
     not_poison &= (b == 0u || a != 0u);
     break;
 
+  case SAdd_Overflow:
+    val = a + b;
+    val = val.concat(expr::mkIf(a.add_no_soverflow(b), expr::mkUInt(0, 1), expr::mkUInt(1, 1)));
+    break;
+
+  case ExtractValue:
+  {
+    auto &aggType = static_cast<AggregateType&>(lhs.getType());
+    val = aggType.extract(a, b);
+
+    uint64_t n;
+    not_poison &= b.isUInt(n);
```

Comment:
is this not_poison bit needed? ENSURE() seems more appropriate since this should always hold by the type checker.

Reply:
typechecker can only enforce the number of bits of the type but what happens if the real value is within the enforced bits but still greater than the total aggregate size? shouldn't it be poison then?

Reply:
fixed